src/libcharon/plugins/kernel_super/kernel_super_ipsec.c

```diff
@@ -2213,6 +2251,83 @@ METHOD(kernel_ipsec_t, destroy, void,
 	free(this);
 }
 
+struct tunnel_counter_key {
+	uint32_t terminator_id;
+	const char* tenant_id;
+	const char* account_id;
+	const char* vrf_name;
+};
+
+static bool tunnel_counter_key_compare(const void *key, const void *other_key) {
+    struct tunnel_counter_key *tunnel_1 = (struct tunnel_counter_key *)key;
+    struct tunnel_counter_key *tunnel_2 = (struct tunnel_counter_key *)other_key;
+
+	return tunnel_1->terminator_id == tunnel_2->terminator_id &&
+			!strcmp(tunnel_1->tenant_id, tunnel_2->tenant_id) && 
+			!strcmp(tunnel_1->account_id, tunnel_2->account_id) && 
+			!strcmp(tunnel_1->vrf_name, tunnel_2->vrf_name);
+}
+
+static u_int tunnel_counter_key_hash(const void *key) {
+	struct tunnel_counter_key *tunnel_key = (struct tunnel_counter_key *)key;
+	char tunnel_key_str[MAX_ACCOUNT_ID_LEN + MAX_TENANT_ID_LEN + MAX_VRF_NAME_LEN + 10];
+	sprintf(tunnel_key_str, "%u%s%s%s", tunnel_key->terminator_id, tunnel_key->tenant_id, tunnel_key->account_id, tunnel_key->vrf_name);
+	return hashtable_hash_str(&tunnel_key_str);
+}
+
+static bool count_tunnels(private_kernel_super_ipsec_t* this, struct pm_terminator_conn* const terminator_conn) {
+	const struct pm_proto_addr lookup_addr = *picomsg_terminator_conn_get_lookup_addr(terminator_conn);
+	uint32_t term_id = picomsg_terminator_conn_get_id(terminator_conn);
+	enumerator_t* enumerator;
+	ike_sa_t* ike_sa;
+	uint64_t sa_counter = 0;
+	hashtable_t *counter_map = NULL;
+
+	counter_map = hashtable_create(tunnel_counter_key_hash, tunnel_counter_key_compare, 10);
+
+	enumerator = charon->controller->create_ike_sa_enumerator(charon->controller, true);
+	while (enumerator->enumerate(enumerator, &ike_sa)) {
+		struct pm_proto_addr ike_sa_addr;
+		host_addr_to_picomsg_addr(ike_sa->get_my_host(ike_sa), &ike_sa_addr);
+		if (pm_proto_addr_equals(&ike_sa_addr, &lookup_addr)) {
+			struct super_cert_info cert_info;
+			if (! parse_peer_certificate(ike_sa, &cert_info, this->peer_cert_info_local)) {
+				LOG(DBG_KNL, SL_LOG_CRIT, "failed parsing peer certificate information");
+				enumerator->destroy(enumerator);
+				return false;
+			}
+
+			struct tunnel_counter_key curr_key = {term_id, cert_info.tenant_id, cert_info.account_id, cert_info.vrf_name};
+			void *curr_val = counter_map->get(counter_map, &curr_key);
+			counter_map->put(counter_map, &curr_key, (char*)curr_val + 1);
+
+			++sa_counter;
```

Comment:
In case of IKE rekeying or make-before-break reauth then a single peer will have multiple IKE SAs, so this loop can double count the number of tunnels.
The only way to truly count the number of tunnels and not the number of IKE SAs need to also keep track of the public address of the peer and take it into account (multiple IKE SAs from the same public address should count as 1 tunnel).
This however will complicate things here which I think is unnecessary.
So I suggest you check if can live with potential double counting of tunnels in case of rekey/reauth and update the requirements accordingly.